diff -rupN gimagereader-2.99/configure.ac gimagereader-2.99-new/configure.ac
--- gimagereader-2.99/configure.ac	2014-02-10 03:07:22.000000000 +0100
+++ gimagereader-2.99-new/configure.ac	2014-02-10 12:50:23.359124414 +0100
@@ -33,7 +33,10 @@ LT_INIT
 
 dnl ********************************* Modules *********************************
 PKG_CHECK_MODULES(GIMAGEREADER, [gtkmm-3.0 >= 3.6 gtkspellmm-3.0 >= 3.0.2
-								 cairomm-1.0 poppler-glib tesseract >= 3.02.02])
+								 cairomm-1.0 poppler-glib])
+GIMAGEREADER_CFLAGS+=-I/usr/include/tesseract
+GIMAGEREADER_LIBS+=-ltesseract
+
 
 dnl *************************** Enable version check **************************
 AC_ARG_ENABLE([versioncheck],
